junifer/api/utils.py

```diff
@@ -0,0 +1,124 @@
+"""Provide utility functions for the api sub-package."""
+
+# Authors: Synchon Mandal <s.mandal@fz-juelich.de>
+# License: AGPL
+
+import os
+import platform as pl
+from typing import Dict
+
+from .._version import __version__
+from ..utils.logging import get_versions
+
+
+def _get_junifer_version() -> Dict[str, str]:
+    """Get junifer version information.
+
+    Returns
+    -------
+    dict
+        A dictionary containing junifer version.
+
+    """
+    return {
+        "version": __version__,
+    }
+
+
+def _get_python_information() -> Dict[str, str]:
+    """Get installed Python information.
+
+    Parameters
+    ----------
+    dict
+        A dictionary containing Python information.
+
+    """
+    return {
+        "version": pl.python_version(),
+        "implementation": pl.python_implementation(),
+    }
+
+
+def _get_dependency_information(long_: bool) -> Dict[str, str]:
+    """Get Python environment dependency information.
+
+    Parameters
+    ----------
+    long_ : bool
+        Whether to report long version.
+
+    Returns
+    -------
+    dict
+        A dictionary containing Python dependency information.
+
+    """
+    dependency_versions = get_versions()
+
+    pruned_dependency_versions = {}
+    # Report long version
+    if long_:
+        for key, value in dependency_versions.items():
+            # Ignore built-in modules and self
+            if value != "None" and key != "junifer":
+                pruned_dependency_versions[key] = value
+
+    # Report short version
+    else:
+        for key in [
```

Comment:
Can we read this from the pyproject.toml file?

Or we need to manually hardcode here?

I have the impression that it will end-up like the `_version.py` file, which is created on build/install

Reply:
The only reason I want to avoid reading from `pyproject.toml` is because of the extra I/O. The dependencies are not going to change frequently so hardcoding it doesn't seem like a bad idea.

I don't really get the analogy with the `_version.py` though.

Reply:
On build/install time, a file `_dependencies.py` is created with an array that contains the dependencies.

Soon, we will start having "conditional" dependencies in which some specific markers will require some dependencies. I'm just trying to avoid having always in mind that we need to manage dependencies in two different sites.

Reply:
e.g.: `pip install junifer[mri]` `pip install junifer[eeg]` `pip install junifer[surf]` `pip install junifer[dmri]` ... etc

Reply:
> On build/install time, a file `_dependencies.py` is created with an array that contains the dependencies.
> 
> Soon, we will start having "conditional" dependencies in which some specific markers will require some dependencies. I'm just trying to avoid having always in mind that we need to manage dependencies in two different sites.

I don't remember seeing `_dependencies.py` but will check it.

Reply:
Now implemented with an improved regex search.